include/lsst/sphgeom/codec.h

```diff
@@ -36,32 +36,32 @@ namespace sphgeom {
 /// `encode` appends an IEEE double in little-endian byte order
 /// to the end of buffer.
 inline void encodeDouble(double item, std::vector<uint8_t> & buffer) {
-#if defined(__x86_64__)
-    // x86-64 is little endian.
+#if defined(__x86_64__) or defined(__aarch64__)
+    // x86-64 and Apple Arm64 are little endian.
```

Comment:
I don't think `__aarch64__` is Apple specific, and AArch64 is not guaranteed to be little-endian: https://developer.arm.com/documentation/102376/0100/Alignment-and-endianness. I'm unsure whether big-endian implementations are a real concern, but maybe there's a preprocessor check for little-endianness that could be added to the one for `__aarch64__` (or failing that, `__APPLE__`)?

Also, it might be good to provide a way to force the fallback path to get used (like NO_SIMD in other parts of the code), so it can be tested without exotic hardware or cross-compilation.

Reply:
Hi @smonkewitz, both points are good. I've now gone down the rabbit hole of trying to find which compilers on which architectures define a macro that would be useful here, and it looks like at least Apple defines `__LITTLE_ENDIAN__` on arm64, so that should handle the most relevant case today without being over-inclusive.